cmd/sourced/compose/workdir/common.go

```diff
@@ -29,113 +26,60 @@ var (
 	ErrMalformed = goerrors.NewKind("workdir %s is not valid: %s")
 )
 
-// Init creates a working directory in ~/.sourced for the given repositories
-// directory. The working directory will contain a docker-compose.yml and a
-// .env file.
-// If the directory is already initialized the function returns with no error.
-// The returned value is the absolute path to $HOME/.sourced/workdirs/reposdir
-func Init(reposdir string) (string, error) {
-	defaultFilePath, err := composefile.InitDefault()
-	if err != nil {
-		return "", err
-	}
-
-	workdir, err := path(reposdir)
-	if err != nil {
-		return "", err
-	}
-
-	err = os.MkdirAll(workdir, 0755)
-	if err != nil {
-		return "", errors.Wrap(err, "could not create working directory")
-	}
-
-	composePath := filepath.Join(workdir, "docker-compose.yml")
-	_, err = os.Stat(composePath)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return "", errors.Wrap(err, "could not read the existing docker-compose.yml file")
-		}
-
-		err = os.Symlink(defaultFilePath, composePath)
-		if err != nil {
-			return "", errors.Wrap(err, "could not create symlink to docker-compose.yml file")
-		}
-	}
-
-	envPath := filepath.Join(workdir, ".env")
-	emptyFile, err := isEmptyFile(envPath)
-	if err != nil {
-		return "", errors.Wrap(err, "could not read .env file contents")
-	}
-
-	if emptyFile {
-		hash := sha1.Sum([]byte(reposdir))
-		hashSt := hex.EncodeToString(hash[:])
-
-		contents := fmt.Sprintf(
-			`GITBASE_REPOS_DIR=%s
-COMPOSE_PROJECT_NAME=srcd-%s
-`,
-			reposdir, hashSt)
-
-		err = ioutil.WriteFile(envPath, []byte(contents), 0644)
-		if err != nil {
-			return "", errors.Wrap(err, "could not write .env file")
-		}
-	}
-
-	return workdir, nil
+type envFile struct {
+	Workdir             string
+	ReposDir            string
+	GithubOrganizations []string
+	GithubToken         string
 }
 
-// isEmptyFile returns true if the file does not exist or if it exists but
-// contains empty text
-func isEmptyFile(path string) (bool, error) {
-	_, err := os.Stat(path)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return false, err
-		}
-
-		return true, nil
-	}
-
-	contents, err := ioutil.ReadFile(path)
-	if err != nil {
-		return false, err
-	}
-
-	strContents := string(contents)
-	return strings.TrimSpace(strContents) == "", nil
+func (f *envFile) String() string {
+	volumeType := "bind"
+	volumeSource := f.ReposDir
+	gitbaseSiva := ""
+	if f.ReposDir == "" {
+		volumeType = "volume"
+		volumeSource = "gitbase_repositories"
+		gitbaseSiva = "true"
+	}
+
+	return fmt.Sprintf(`COMPOSE_PROJECT_NAME=srcd-%s
+	GITBASE_VOLUME_TYPE=%s
+	GITBASE_VOLUME_SOURCE=%s
+	GITBASE_SIVA=%s
+	GITHUB_ORGANIZATION=%s
+	GITHUB_TOKEN=%s
```

Comment:
I think we could omit `GITBASE_SIVA`, `GITHUB_ORGANIZATION`, `GITHUB_TOKEN` when not in an org `.env`

Reply:
the idea is not to use `${var:-}` in `docker-compose.yml` so it easy to spot if something is broken (.env file and yml file aren't in sync). It complains if there is nothing set in such case.
And to avoid warning for a normal workflow we set env vars into empty values.

Though I don't feel strongly about it. We can change later if everybody prefers.